docker/Dockerfile

```diff
@@ -0,0 +1,52 @@
+FROM node:12-alpine as host
+LABEL maintainer="Cyril Selasi <cyrilselasi@ymail.com>"
+
+RUN apk add --no-cache --virtual .gyp python make g++
+
+FROM host as base
+
+ENV SERVER_ROOT /usr/src/app
+RUN mkdir -p ${SERVER_ROOT}
+
+WORKDIR ${SERVER_ROOT}
+COPY . .
+
+RUN yarn install --ignore-optional --link-duplicates
+RUN apk del .gyp && rm -rf /var/cache/apk/*
+
+RUN yarn build
+
+
+# ---------------------------------------------
+
+# Production vendoring stage
+FROM host as node_modules
+
+WORKDIR /tmp/
+COPY --from=base /usr/src/app/package.json /usr/src/app/yarn.lock ./
+
+RUN yarn install --frozen-lockfile --ignore-optional --production --link-duplicates
+RUN apk del .gyp && rm -rf /var/cache/apk/*
+
+# ---------------------------------------------
+
+## Final composition stage
+FROM alpine:3 as production
+RUN apk add npm && rm -rf /var/cache/apk/*
```

Comment:
Is there any extra benefit by using `alpine:*` instead of `node:*-alpine` as the final base image?

Reply:
Actually, I usually use the `node:*-alpine` images as the final stage. It has a number of verifications in its definition to verify the origin.

The image size will come down to about the same so there is not a huge advantage there. The `node:*-alpine` image should be preferred.